### PR TITLE
an overload for GetWindowPlacement + new helper TryGetWindowPlacement

### PR DESCRIPTION
**Pull request by lostmsu**
Slightly easier to use variants of GetWindowPlacement

### CODE REVIEW

File: src/User32/User32.cs [hunk 1]
```diff
@@ -2498,6 +2498,21 @@ public static partial class User32
...
+        /// Before calling GetWindowPlacement, set the length member to sizeof(WINDOWPLACEMENT). GetWindowPlacement fails if
+        /// lpwndpl-> length is not set correctly.
+        /// </param>
+        /// <returns>
+        /// If the function succeeds, the return value is nonzero.
+        /// <para>If the function fails, the return value is zero. To get extended error information, call GetLastError.</para>
+        /// </returns>
+        [DllImport(nameof(User32), SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool GetWindowPlacement(IntPtr hWnd, ref WINDOWPLACEMENT lpwndpl);
```
Comment (AArnott) [inline src/User32/User32.cs:2514]: We don't typically add multiple overloads manually like this. We rely on our codegen tool to generate the pointer-free overloads. Did that not work for you?

File: src/User32/User32.Helpers.cs [hunk 2]
```diff
@@ -369,5 +369,18 @@ public static unsafe WINDOWPLACEMENT GetWindowPlacement(IntPtr hWnd)
...
+        /// <param name="hWnd">A handle to the window.</param>
+        /// <returns>
+        /// A WINDOWPLACEMENT structure with the show state and position information.
+        /// If the data can not be retrieved, <c>null</c> is returned. In that case <see cref="GetLastError"/>
+        /// or <see cref="Win32Exception"/> can be used to obtain the error.
+        /// </returns>
+        public static unsafe WINDOWPLACEMENT? TryGetWindowPlacement(IntPtr hWnd)
+        {
+            var result = WINDOWPLACEMENT.Create();
+            return GetWindowPlacement(hWnd, &result) ? (WINDOWPLACEMENT?)result : null;
```
Comment (AArnott) [inline src/User32/User32.Helpers.cs:383]: We don't usually define methods like this. Per our [guidelines on helper methods](https://github.com/AArnott/pinvoke/blob/master/CONTRIBUTING.md#helper-methods) I think this falls out of scope.

@vbfox what do you think?

**Comment by lostmsu** [comment]
@AArnott sorry, looks like I should have read CONTRIBUTING

**Comment by AArnott** [comment]
Thanks for sharing anyway, @lostmsu. I look forward to your next PR.

**Pull request closed without merging**